src/super_gradients/training/datasets/detection_datasets/coco_detection_yolox.py

```diff
@@ -0,0 +1,287 @@
+import numpy as np
+
+from super_gradients.training.utils.detection_utils import get_yolox_datadir
+
+logger = get_logger(__name__)
+
+
+class COCODetectionDatasetYolox(Dataset):
+    """Detection dataset COCO implementation"""
+
+    def __init__(
+            self, img_size: tuple = (640, 640),
+            data_dir: str = None,
+            json_file: str = "instances_train2017.json",
+            name: str = "images/train2017",
+            cache: bool = False,
+            tight_box_rotation: bool = False,
+            transforms: list = [],
+    ):
+        """
+        
+        :param img_size: tuple, Image size (when loaded, before transforms)
+        :param data_dir: str, root path to coco data.
+        :param json_file: str, path to coco json file.
+        :param name: str, sub directory of data_dir containing the data.
+        :param cache: bool, whether to cache images
+        :param tight_box_rotation: bool, whether to use of segmentation maps convex hull
+         as target_seg (see load_sample docs).
+        :param transforms: list of transforms to apply sequentially on sample in __getitem__
+        """
+        super().__init__()
+        self.imgs = None
+        self.data_dir = data_dir
+        self.input_dim = img_size
+        if data_dir is None:
+            data_dir = os.path.join(get_yolox_datadir(), "COCO")
+        self.data_dir = data_dir
+        self.json_file = json_file
+        self.name = name
+        self.coco = COCO(os.path.join(self.data_dir, "annotations", self.json_file))  # duplicate
+        self.tight_box_rotation = tight_box_rotation
+        remove_useless_info(self.coco, self.tight_box_rotation)
+        self.ids = self.coco.getImgIds()
+        self.class_ids = sorted(self.coco.getCatIds())
+        cats = self.coco.loadCats(self.coco.getCatIds())
+        self._classes = tuple([c["name"] for c in cats])
+        self.annotations = self._load_coco_annotations()
+        if cache:  # cache after merged
+            self._cache_images()
+
+        self.transforms = transforms
+
+    def __getitem__(self, idx):
+        sample = self.load_sample(idx)
+        sample = self.apply_transforms(sample)
+        return sample["image"], sample["target"], sample["info"], sample["id"]
+
+    def __len__(self):
+        return len(self.ids)
+
+    def _load_coco_annotations(self):
+        return [self._load_anno_from_ids(_ids) for _ids in self.ids]
+
+    def _load_anno_from_ids(self, id_):
+        im_ann = self.coco.loadImgs(id_)[0]
+        width = im_ann["width"]
+        height = im_ann["height"]
+        anno_ids = self.coco.getAnnIds(imgIds=[int(id_)], iscrowd=False)
+        annotations = self.coco.loadAnns(anno_ids)
+        objs = []
+        for obj in annotations:
+            x1 = np.max((0, obj["bbox"][0]))
+            y1 = np.max((0, obj["bbox"][1]))
+            x2 = np.min((width, x1 + np.max((0, obj["bbox"][2]))))
+            y2 = np.min((height, y1 + np.max((0, obj["bbox"][3]))))
```

Comment:
what is the logic here? we don't trust the annotations file to be accurate? x2-x1 can be different than width?

Reply:
Yes, this happens sometimes.